--- a/java/common/rules/java_single_jar.bzl
+++ b/java/common/rules/java_single_jar.bzl
@@ -15,6 +15,7 @@
 
 load("//java/common:java_common.bzl", "java_common")
 load("//java/common:java_info.bzl", "JavaInfo")
+load("//java/common:java_semantics.bzl", "semantics")
 
 # copybara: default visibility
 
@@ -77,13 +78,14 @@
     if ctx.attr.exclude_pattern:
         args.add("--exclude_pattern", ctx.attr.exclude_pattern)
 
+    toolchain = semantics.find_java_toolchain(ctx)
     ctx.actions.run(
         inputs = inputs,
         outputs = [ctx.outputs.output],
         arguments = [args],
         progress_message = "Merging into %s" % ctx.outputs.output.short_path,
         mnemonic = "JavaSingleJar",
-        executable = ctx.executable._singlejar,
+        executable = toolchain.single_jar,
         use_default_shell_env = True,
     )
 
@@ -131,15 +133,10 @@
         "exclude_pattern": attr.string(default = "", doc = """
             A regex pattern of files to exclude from the jar.
         """),
-        "_singlejar": attr.label(
-            default = Label("//toolchains:singlejar"),
-            cfg = "exec",
-            allow_single_file = True,
-            executable = True,
-        ),
         "output": attr.output(),
     },
     implementation = _bazel_java_single_jar_impl,
+    toolchains = [semantics.JAVA_TOOLCHAIN],
     doc = """
 Collects Java dependencies and jar files into a single jar
 
