--- a/default_java_toolchain.bzl
+++ b/default_java_toolchain.bzl
@@ -15,4 +15,6 @@
 """Bazel rules for creating Java toolchains."""
 
+load("@rules_java//java:defs.bzl", "java_common", "java_toolchain")
+
 # JVM options, without patching java.compiler and jdk.compiler modules.
 BASE_JDK9_JVM_OPTS = [
@@ -147,3 +149,3 @@
     toolchain_args.update(kwargs)
-    native.java_toolchain(
+    java_toolchain(
         name = name,
