--- a/java/common/rules/java_single_jar.bzl
+++ b/java/common/rules/java_single_jar.bzl
@@ -86,13 +86,14 @@
     if ctx.attr.exclude_pattern:
         args.add("--exclude_pattern", ctx.attr.exclude_pattern)
 
+    toolchain = semantics.find_java_toolchain(ctx)
     ctx.actions.run(
         inputs = depset(build_info_files, transitive = [inputs]),
         outputs = [ctx.outputs.output],
         arguments = [args],
         progress_message = "Merging into %s" % ctx.outputs.output.short_path,
         mnemonic = "JavaSingleJar",
-        executable = ctx.executable._singlejar,
+        executable = toolchain.single_jar,
         use_default_shell_env = True,
     )
 
@@ -140,12 +141,6 @@
         "exclude_pattern": attr.string(default = "", doc = """
             A regex pattern of files to exclude from the jar.
         """),
-        "_singlejar": attr.label(
-            default = Label("//toolchains:singlejar"),
-            cfg = "exec",
-            allow_single_file = True,
-            executable = True,
-        ),
         "output": attr.output(),
         "stamp": attr.int(
             doc = """
@@ -163,6 +158,7 @@
         "_build_info_translator": attr.label(default = semantics.BUILD_INFO_TRANSLATOR_LABEL),
     },
     implementation = _bazel_java_single_jar_impl,
+    toolchains = [semantics.JAVA_TOOLCHAIN],
     doc = """
 Collects Java dependencies and jar files into a single jar
 
